lib/bindevict.go

```diff
@@ -116,10 +119,16 @@ func (entry *BindThrottle) incrAllowEveryX() {
 	}
 }
 
-func (be *BindEvict) ShouldBlock(sqlhash uint32, bindKV map[string]string, heavyUsage bool) (bool, *BindThrottle) {
-	GetBindEvict().lock.Lock()
-	sqlBinds := GetBindEvict().BindThrottle[sqlhash]
-	GetBindEvict().lock.Unlock()
+func (bindEvict *BindEvict) ShouldBlock(sqlhash uint32, bindKV map[string]string, heavyUsage bool) (bool, *BindThrottle) {
+	entryTime := time.Now()
+	defer func() {
+		if logger.GetLogger().V(logger.Verbose) {
```

Comment:
Do we need this info?

Reply:
During testing, this is to understand how much time the call to ShouldBlock takes, even though we disable bind throttle.

Reply:
We should avoid more I/O and calculating time difference at this heavily used function call. If really needed, atomic cal txn is better.

Reply:
reverted the change now